### PR TITLE
fix(curriculum): Dice Game - Missing some small straights

### LINKED ISSUE #54079: Dice Game - Missing some small straights
### Describe the Issue

You can get a small straight with `"12346"` but that's excluded.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/step-84

### Your code

```js
  if (smallStraightsArr.some(straight => uniqueNumbersStr.includes(straight))) {
    updateRadioOption(3, 30);
  }
```

### Expected behavior

All valid small straights captured.

We should put Step 85 before the current Step 84, then update Step 84 to use something like the logic given above.

### Forum Post

https://forum.freecodecamp.org/t/learn-intermediate-algorithmic-thinking-by-building-a-dice-game-step-87/679303

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Nitz2611**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54079

### CODE REVIEW

**Comment by Nitz2611** [comment]
I switched steps 84 and 85, which ended up disrupting the order of steps in the UI. Can anyone 
messedup
please help me correct this?

**Review by gikf — COMMENTED** [review]
When changing order in the _files_, the order in the meta file also needs to be changed - https://github.com/Nitz2611/freeCodeCamp/blob/issue#54079/curriculum/challenges/_meta/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/meta.json

**Comment by Nitz2611** [comment]
Thank you for your help.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md:10]: numbers need to be wrapped in backticks. revert this back to the way it was

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md:19]: The issue is that we are expecting the camper to arrive at this solution.
However the directions provide no indication that the `some` method needs to be used here or that the callback for the `some` method need a parameter of `straight`. 

This test has multiple correct answers.
If you look at the original [forum post](https://forum.freecodecamp.org/t/learn-intermediate-algorithmic-thinking-by-building-a-dice-game-step-87/679303/5), they originally came up with this code 

```js
smallStraightArr.forEach(straight => {
  if(uniqueNumberStr.includes(straight)) {
    updateRadioOption(3, 30);
  }
}
```

so we should look into ways to allow for multiple valid answers and write a more flexible test that don't involve a regex like this

Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md:19]: The `some` method is a better approach because the `updateRadioOption(3, 30);` function would only happen once. 

If we do want the campers to only arrive at this solution, then the directions need to be update to do a better job of leading them there.
Also the hints would need to better illustrate the solution.
Otherwise campers will struggle to get there

**Comment by jdwilkin4** [comment]
I have added the blocked label on this for now, because we should decide if we want to have just one solution using the `some` method or all for campers to use multiple approaches. 

I will go ahead and update the issue as well to remove the help wanted label for now so a better consensus can be reached.

**Comment by jdwilkin4** [comment]
I removed the blocked label. 
We are going to go with approach no.2 where multiple valid approaches should be accounted for in the tests.

**Comment by Nitz2611** [comment]
Hey @jdwilkin4, can you guide me how to implement the approach you suggested, so that i can make neccessery changes required.

**Comment by Nitz2611** [comment]
I just want to know that how to change test regex that accepts multiple solutions.

**Comment by jdwilkin4** [comment]
@Nitz2611 

A regex isn't going to work here because it will be very messy to account for multiple solutions.
I like the approach done in the other open PR here.
https://github.com/freeCodeCamp/freeCodeCamp/pull/54138

I haven't been able to test it out yet, but I like that approach because it works, then the camper can use whatever loop approach they want

**Comment by JoyShaheb** [comment]
@Nitz2611  buddy give this a quick try ?
image
also, i updated the [issue ticket](https://github.com/freeCodeCamp/freeCodeCamp/issues/54079) with some details/ suggestions, give it a quick try and let me know ?

**Comment by Nitz2611** [comment]
Oh my bad. I am sorry

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md:12]: i suggest making the text shorter and simpler ?

```suggestion
To get started, iterate through `smallStraightsArr` using any loop you prefer. If using a higher-order function like `forEach` or `some`, the callback function should take an argument of `straight`. If using `for...of`, the loop variable should also be `straight`.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md:21]: the test doesn't pass when there's some spaces before and after the arguments
```suggestion
assert.match(code, /\s*i?f?\(\s*uniqueNumbersStr\.includes\(\s*(straight|smallStraightsArr\[i\])\s*\)\s*\)\s*{/)
```

**Review by JoyShaheb — CHANGES_REQUESTED** [review]
dropped some reviews !

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md:14]: doesn't this sound correct ?
```suggestion
Inside the loop,  if `uniqueNumbersStr` has a combination that matches a small straight, call the `updateRadioOption` with `3` and `30` as the arguments.
```

**Review by JoyShaheb — CHANGES_REQUESTED** [review]

**Review by Ksound22 — COMMENTED** [review]
This is okay with me. I will gladly approve when the requested change is implemented.

**Comment by Nitz2611** [comment]
Hey @Ksound22 
The regex suggested by @JoyShaheb is not working and also not passing some automated tests. The regex update commited by me is working fine.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/657e2ecee9f60092c89338d9.md:21]: why is there a question mark in between the `if` word ? try this suggestion ?
```suggestion
assert.match(code, /if\s*\(\s*uniqueNumbersStr\.includes\(\s*(straight|smallStraightsArr\[i\])\s*\)\s*\)\s*{/)
```

**Review by JoyShaheb — CHANGES_REQUESTED** [review]

**Comment by JoyShaheb** [comment]
this [PR](https://github.com/freeCodeCamp/freeCodeCamp/pull/54138) got accepted earlier, unfortunately we have to close this issue, Thank you for contributing 🙏

**Pull request closed without merging**
